apps/web/src/services/dashboards/widgets/asset-widgets/severity-status-by-service/SeverityStatusByServiceWidget.vue

```diff
@@ -38,48 +39,115 @@ import {
 const state = reactive({
-    ...toRefs(useWidgetState(props)),
+    ...toRefs(useWidgetState<Data[]>(props)),
     dateRange: computed<DateRange>(() => ({
         start: dayjs.utc(state.settings?.date_range?.start).format(DATE_FORMAT),
         end: dayjs.utc(state.settings?.date_range?.end).format(DATE_FORMAT),
     })),
-    sampleData: computed(() => [...Array(20)].map(() => Object.values(SEVERITY_STATUS_MAP)[Math.floor(Math.random() * 5)])),
     boxWidth: computed<number>(() => {
         if (!props.width) return 112;
         const widgetPadding = 24;
         const widgetContentWidth = props.width - (widgetPadding * 2);
         if (props.width >= 990) return widgetContentWidth / 8;
-        return widgetContentWidth / 7;
+        return widgetContentWidth / 7 < 112 ? 112 : widgetContentWidth / 7;
     }),
 });
 const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
 
-const fetchData = async () => {
-    //
+
+/* Api */
+const fetchData = async (): Promise<Data[]> => {
+    try {
+        const apiQueryHelper = new ApiQueryHelper();
+        apiQueryHelper
+            .setFilters(state.consoleFilters)
+            .addFilter({ k: 'ref_cloud_service_type.labels', v: 'Compliance', o: '=' });
+        const results = await SpaceConnector.clientV2.inventory.cloudService.analyze({
+            query: {
+                group_by: ['data.service'],
+                fields: {
+                    status: {
+                        key: 'data.status',
+                        operator: 'add_to_set',
+                    },
+                    severity: {
+                        key: 'data.severity',
+                        operator: 'add_to_set',
+                    },
+                    fail_finding_count: {
+                        key: 'data.stats.checks.fail',
+                        operator: 'sum',
+                    },
+                },
+                sort: [{
+                    key: 'service',
+                    desc: false,
+                }],
+                page: {
+                    limit: 80,
+                    start: 1,
+                },
+                ...apiQueryHelper.data,
+            },
+        });
+        return refineData(results);
+    } catch (e) {
+        console.error(e);
+        return [];
+    }
+};
+
+
+/* Util */
+const refineData = (data): Data[] => {
+    const { results } = data;
+    return results.map((result) => {
+        const minSeverityPriority = min(result.severity.map((s) => SEVERITY_STATUS_MAP[s].priority)); // ex. 1
+        const severity = Object.values(SEVERITY_STATUS_MAP)
+            .filter((s) => s.priority === minSeverityPriority)[0].name; // ex. 'CRITICAL'
+        const complianceStatus = result.status.includes('FAIL') ? 'FAIL' : 'PASS';
+        return {
+            service: result.service,
+            severity,
+            complianceStatus,
+            value: result.fail_finding_count,
+        };
+    });
 };
```

Comment:
nitpick: It was pretty hard to read for me. 🥲
What about separate the process of getting the exact priority's severity name?

```suggestion
const SEVERITY_PRIORITY_MAP: Record<number, typeof SEVERITY_STATUS_MAP[Severity]> = {}
Object.values(SEVERITY_STATUS_MAP).forEach(s => { SEVERITY_STATUS_MAP[s.priority] = s })

const refineData = (data): Data[] => {
    const { results } = data;
    return results.map((result) => {
        const minSeverityPriority = min(result.severity.map((s) => SEVERITY_STATUS_MAP[s].priority)); // ex. 1
        const severity = SEVERITY_PRIORITY_MAP[minSeverityPriority]?.name // ex. 'CRITICAL'
        const complianceStatus = result.status.includes('FAIL') ? 'FAIL' : 'PASS';
        return {
            service: result.service,
            severity,
            complianceStatus,
            value: result.fail_finding_count,
        };
    });
};
```